File: src/market-tracker-webapi/Application/Service/Operations/List/IListEntryService.cs
using market_tracker_webapi.Application.Domain.Filters.List;
using market_tracker_webapi.Application.Domain.Schemas.List;
using market_tracker_webapi.Application.Service.Results;

namespace market_tracker_webapi.Application.Service.Operations.List;

public interface IListEntryService
{
    Task<ShoppingListEntriesResult> GetListEntriesAsync(string listId, Guid clientId,
        ShoppingListAlternativeType? alternativeType,
        IList<int>? companyIds,
        IList<int>? storeIds,
        IList<int>? cityIds);

    Task<ListEntryId> AddListEntryAsync(string listId, Guid clientId, string productId,
        int storeId, int quantity);

    Task<ListEntry> UpdateListEntryAsync(string listId, Guid clientId, string entryId,
        int? storeId, int? quantity);

    Task<ListEntryId> DeleteListEntryAsync(string listId, Guid clientId, string entryId);
}

File: src/market-tracker-android/app/src/main/java/pt/isel/markettracker/http/service/operations/list/listEntry/IListEntryService.kt
package pt.isel.markettracker.http.service.operations.list.listEntry

import pt.isel.markettracker.domain.model.list.listEntry.ListEntry
import pt.isel.markettracker.domain.model.list.listEntry.ShoppingListEntries

interface IListEntryService {

    suspend fun addListEntry(
        listId: String,
        productId: String,
        storeId: Int,
        quantity: Int,
    ): String

    suspend fun updateListEntry(
        listId: String,
        entryId: String,
        storeId: Int,
        quantity: Int,
    ): ListEntry

    suspend fun deleteListEntry(listId: String, entryId: String)

    suspend fun getListEntries(
        listId: String,
        alternativeType: AlternativeType? = null,
        companyIds: List<Int> = emptyList(),
        storeIds: List<Int> = emptyList(),
        cityIds: List<Int> = emptyList(),
    ): ShoppingListEntries
}